pandas/io/common.py

```diff
@@ -382,6 +382,19 @@ def _get_filepath_or_buffer(
         # urlopen function defined elsewhere in this module
         import urllib.request
 
+        # Fix for GH #55828
+        parsed_url = parse_url(filepath_or_buffer)
```

Comment:
I believe `is_url` should not be true for fsspec urls. So that might be a much nicer way of fixing this issue (I think @krehm was also hinting at that in the issue) - I'm not familiar with the urllib regex, we might need to exclude more fsspec URLs from it.

Reply:
@twoertwein it did seem to me that any fsspec url in is_url is guaranteed to fail in this case, which seemed like a logic flaw to me.  But I'm not familiar with the urllib code either, so was hesitant to specify a particular solution.

Reply:
The overlap between urllib/fsspec is at the moment:
```py
>>> set(uses_relative + uses_netloc + uses_params).intersection(fsspec.available_protocols())
{'sftp', 'ftp', 'file', 'git', 'http', 'https'}
```

Could have something like this:
```py
_VALID_URLS = set(uses_relative + uses_netloc + uses_params).difference( fsspec.available_protocols())
_VALID_URLS.update(("http", "https"))
_VALID_URLS.discard("")
```
Technically this is a behavior change for sftp, git, ... (might be okay, probably not used frequently?). fsspec should have `available_protocols` since early 2022 https://github.com/fsspec/filesystem_spec/pull/913 might need to double check whether we need to bump the minimum version of fsspec. This might make the regex in `is_fsspec_url` obsolete.

@mroeschke

Reply:
@twoertwein I didn't want to touch `is_url` as its used in a few other places and I wasn't sure if it would break anything. Is it okay to do so?

These are the places is_url is used without a corresponding is_fsspec_url call:
```
pandas.io.html._LxmlFrameParser._build_doc
pandas.io.html._read
pandas.io.formats.html.HTMLFormatter._write_cell
```